frontend/src/components/employees/index.tsx

```diff
@@ -0,0 +1,145 @@
+
+/**
+ * Functions which extract the data to be displayed in the table columns
+ * from StaffData.
+ */
+const getName = ({ firstName, surname }: StaffData) =>
+  `${firstName} ${surname}`;
+const getJobTitle = ({ jobTitle }: StaffData) => jobTitle;
+const partFullTime = (fullTimeHours: number) => ({
+  contractedHours,
+}: StaffData) => (contractedHours >= fullTimeHours ? 'full' : 'part');
+const editUserButton = (editFunction: (o: number) => () => any) => ({
+  id,
+}: StaffData) => (
+  <button
+    type="button"
+    className={employeesStyle.editButton}
+    onClick={editFunction(id)}>
+    &#x1F589;
+  </button>
+);
+const removeUserButton = (removeFunction: (o: number) => () => any) => ({
+  id,
+}: StaffData) => (
+  <button
+    type="button"
+    className={employeesStyle.removeButton}
+    onClick={removeFunction(id)}>
+    X
+  </button>
+);
+
+/**
+ * Specifies the list of table columns to be displayed.
+ * Each  column has the following properties:
+ * id - column's unique id (also its class name for css)
+ * name - column name - the top row of the table
+ * content - a function taking StaffData as an argument
+ *    and returning the value of the corresponding table
+ *    cell
+ */
+const TABLE_COLUMNS = [
+  { id: 'name', name: 'Name', content: getName },
+  { id: 'job', name: 'Job title', content: getJobTitle },
+  {
+    id: 'partfull',
+    name: 'Part/Full Time',
+    content: partFullTime(FULLTIME_HOURS),
+  },
+  { id: 'editbtn', name: '', content: editUserButton(editUser) },
+  { id: 'removebtn', name: '', content: removeUserButton(deleteUser) },
+];
+
+const renderTableHeaders = (tableColumns: TableColumn[]) => (
+  <TableHead>
+    <TableRow>
+      {tableColumns.map(({ id, name }: TableColumn) => (
+        <TableCell key={id} className={employeesStyle[id]}>
+          {name}
+        </TableCell>
+      ))}
+    </TableRow>
+  </TableHead>
+);
+
+const renderTableRow = (tableColumns: TableColumn[]) => (row: StaffData) => (
+  <TableRow key={row.id}>
+    {tableColumns.map(({ id, content }: TableColumn) => (
+      <TableCell key={id} className={employeesStyle[id]}>
+        {content(row)}
+      </TableCell>
+    ))}
+  </TableRow>
+);
+
+const renderTable = (tableColumns: TableColumn[]) => (data: StaffData[]) => (
+  <Table size="small">
+    {renderTableHeaders(tableColumns)}
+    <TableBody>{data.map(renderTableRow(tableColumns))}</TableBody>
+  </Table>
+);
+
+const addButton = (
+  <Box display="flex" justifyContent="flex-end" alignItems="center">
+    <Button
+      className={employeesStyle.addButton}
+      color="primary"
+      variant="contained"
+      size="small"
+      onClick={addUser}>
+      Add
+    </Button>
+  </Box>
+);
+
+const Employees = () => {
+  const staffList = useSelector(selectStaff);
+  const dispatch = useDispatch();
+
+  // Fetch data when component loads
+  useEffect(() => {
+    dispatch(fetchStaff(AUTHENTICATION_TOKEN, STAFF_FETCH_URL));
```

Comment:
See my ealier comment about moving the url to the staffSlice class. 

The Auth_Token stuff will be refactored once we have the Auth PR merged